src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h

```diff
@@ -263,7 +265,7 @@ class HeatCapacityPotentialMinimize_TopOpt : public ROL::Objective<real_t> {
 
     if(nodal_density_flag_){
       FEM_->compute_adjoint_gradients(design_densities, objective_gradients);
-      gp->scale(-1);
+      gp->scale(-1/initial_heat_capacity_potential);
```

Comment:
Consider renaming gp to be a meaningful variable name.

Reply:
Those names remain consistent with the ROL arguments they represent.

Reply:
In that case, what does gp represent?

Reply:
Pointer to the gradient vector

Reply:
Ah.  In that case, consider adding a comment when it is created noting that, or rename it to be grad_ptr so future us doesn't have to puzzle it out.

Reply:
lets do that in another PR, considering its the same pattern in every single constraint and objective file.

Reply:
Sounds good.  I just know that future knuckle draggers (myself among them) will take psychic damage from two letter variable names when reading the code lol.